### DISCUSSION TITLE
Get the screenshot of the current active window

### DISCUSSION THREAD

**Discussion by JustSimplyKyle** [hyprwm/Hyprland#412] [General]
Currently I'm using this magic command to get the screenshot of the active window
`wayshot -s "$(hyprctl activewindow | sed -n '2p' | sed -e "s/[^0-9]/ /g" | sed -e 's/^[[:space:]]*//') $(hyprctl activewindow | sed -n '3p' | sed -e "s/[^0-9]/ /g" | sed -e 's/^[[:space:]]*//')" --stdout | wl-copy`
Is there anyway to improve this, because in my eyes, this is cursed as hell

**Comment by vaxerski** [comment]
well if you want it only to be better eyecandy, and not faster, use -j for json and then parse with a json parser

**Reply by JustSimplyKyle** [reply]
Thanks so much!
`s=($(hyprctl activewindow -j | jq -c -r '.at,.size | .[]')); wayshot -s "$s" --stdout | wl-copy`
It's still a bit clunky, having to use bash arrays, but it so much cleaner than the above.

**Reply by JustSimplyKyle** [reply]
Or the `wayshot -s "$(hyprctl activewindow -j | jq -c -r '.at,.size | .[]' | tr '\n' ' ')" --stdout | wl-copy` tr way, which is even faster*
*I'm not sure about that tho
